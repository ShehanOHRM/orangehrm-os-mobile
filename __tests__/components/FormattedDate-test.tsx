/*
 * This file is part of OrangeHRM
 *
 * Copyright (C) 2020 onwards OrangeHRM (https://www.orangehrm.com/)
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

import 'react-native';
import React from 'react';
import Text from 'components/DefaultText';
import FormattedDate from 'components/FormattedDate';
import {Provider} from 'react-redux';
import configureStore from 'store/configureStore';
import {render} from 'react-native-testing-library';
const mockStore = configureStore();

describe('components/FormattedDate', () => {
  test('test FormattedDate component', () => {
    const node = render(
      <Provider store={mockStore}>
        <FormattedDate>{'2020-12-20'}</FormattedDate>
      </Provider>,
    ).toJSON();
    expect(node).toMatchSnapshot();
  });

  test('test FormattedDate component: check undefined', () => {
    const node = render(
      <Provider store={mockStore}>
        <FormattedDate>{undefined}</FormattedDate>
      </Provider>,
    ).toJSON();
    expect(node).toMatchSnapshot();
  });

  test('test FormattedDate component: check string', () => {
    const node = render(
      <Provider store={mockStore}>
        <FormattedDate>{'Test'}</FormattedDate>
      </Provider>,
    ).toJSON();
    expect(node).toMatchSnapshot();
  });

  test('test FormattedDate component: check nested', () => {
    const node = render(
      <Provider store={mockStore}>
        <Text style={{color: 'orange'}}>
          <FormattedDate>{'2020-12-20'}</FormattedDate>
        </Text>
        <Text style={{color: 'orange'}}>
          <FormattedDate nested>{'2020-12-20'}</FormattedDate>
          {' to '}
          <FormattedDate nested>{'2020-12-21'}</FormattedDate>
        </Text>
      </Provider>,
    ).toJSON();
    expect(node).toMatchSnapshot();
  });
});
