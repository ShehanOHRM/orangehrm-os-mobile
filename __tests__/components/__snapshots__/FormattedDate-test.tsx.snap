// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/FormattedDate test FormattedDate component 1`] = `
<Text
  dispatch={[Function]}
  style={
    Array [
      Object {
        "color": "#818181",
      },
      undefined,
      undefined,
    ]
  }
>
  Sun, 20 Dec 2020
</Text>
`;

exports[`components/FormattedDate test FormattedDate component: check nested 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#818181",
        },
        undefined,
        Object {
          "color": "orange",
        },
      ]
    }
  >
    <Text
      dispatch={[Function]}
      style={
        Array [
          Object {
            "color": "#818181",
          },
          undefined,
          undefined,
        ]
      }
    >
      Sun, 20 Dec 2020
    </Text>
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#818181",
        },
        undefined,
        Object {
          "color": "orange",
        },
      ]
    }
  >
    <Text
      dispatch={[Function]}
      style={
        Array [
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      Sun, 20 Dec 2020
    </Text>
     to 
    <Text
      dispatch={[Function]}
      style={
        Array [
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      Mon, 21 Dec 2020
    </Text>
  </Text>,
]
`;

exports[`components/FormattedDate test FormattedDate component: check string 1`] = `
<Text
  dispatch={[Function]}
  style={
    Array [
      Object {
        "color": "#818181",
      },
      undefined,
      undefined,
    ]
  }
>
  Test
</Text>
`;

exports[`components/FormattedDate test FormattedDate component: check undefined 1`] = `
<Text
  dispatch={[Function]}
  style={
    Array [
      Object {
        "color": "#818181",
      },
      undefined,
      undefined,
    ]
  }
/>
`;
